tutorials/arvo/ames.md

```diff
@@ -112,40 +115,299 @@ ducts as unique keys.  Stop tracking a peer by sending Ames a
+on bone 14.
+
+`%plea`s and `%boon`s are handled on separate bones so that e.g. sending a large
+`%boon` doesn't stop an additional `%plea` from being received. It is also
+simply cleaner to make each bone one-way. Acks and nacks are very small packets
+as well as integral to controlling the total orderings on `%plea`s and `%boon`s
+and so they are included on these bones as well.
+
+Naxplanations can potentially be very large, as they often contain things like stack
+traces or other crash reports. Thus it is undesirable to have them share a bone
+with `%boon`s and create congestion there. So naxplanations have their own bone,
+and so do acks to packets that make up a naxplanation, as well as the
+message-level naxplanation ack.
+
+### Packet relaying
+
+Here we describes how the Ames network relays packets. We ignore all details
+about UDP, which occurs at a lower layer than Ames.
+
+When a ship first contacts another ship, it may only know its `@p`, but an IP
+address and port are necessary for peer-to-peer communication to occur.
+Thus the initial correspondence between two ships, or additional correspondence
+following a change in IP or port of one of the parties, will need to be relayed
+by an intermediary that knows the IP and port of the receiving ship.
+
+For now packet relaying is handled entirely by galaxies. Every galaxy is
+responsible for knowing the IP and port of every planet underneath it. In the
+future, galaxies will only need to know the star sponsoring a planet, and stars
+will be responsible for the final step of the relay.
+
+The following diagram summarizes the packet creation and forwarding process.
+
+<div style="text-align:center">
+<img src="https://media.urbit.org/docs/arvo/datagram-long.png">
+</div>
+
+We elaborate on the above diagram.
+
+A typical relay will look something like this: `~bacbel-tagfeb` wishes to
+contact `~worwel-sipnum` but does not know the IP address and port at which
+`~worwel-sipnum` resides. Both planets live under `~zod`, which knows both of
+their IP and port numbers by virtue of being the galaxy that both live under.
+
+So `~bacbel-tagfeb` sends a packet addressed to `~worwel-sipnum` to `~zod`.
+
+The packet has the following format:
+ - The standard Ames header described in [header](#header), where the checksum
+   is the `+mug` of the body, and the sender and receiver ship types are `01`,
+   denoting that the sender and receiver are planets. The remainder of the
+   packet is the body.
+ - The body of this packet will be the `@p` of the
+sender `~bacbel-tagfeb`, followed by the `@p` of the receiver `~worwel-sipnum`,
+followed by the payload.
+ - The payload of this packet will be the `+jam` of `[origin content]`. `origin`
+for this initial packet will be `~`, which implies that the packet is
+originating from the sender. `content` is the `+jam` of the message, encrypted
+using a Diffie-Hellman symmetric key obtained with the private key of
```

Comment:
I think this phrasing is a bit ambiguous.  The symmetric key is obtained using a Diffie-Hellman, once, *and then* the encryption of each packet is performed with that symmetric key.

Reply:
https://github.com/urbit/docs/pull/957/commits/3f725698e167a6d4ff1cf726afbe515cb322a755#diff-b229785bff07fd288a75269882aa1977e802cea1e525ebcf42f4651b428d67daL351-R366